elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/EntityHydrator.java

```diff
@@ -98,11 +101,24 @@ protected Object coerceObjectToEntity(Map<String, Object> result, MutableInt cou
         }
 
         result.forEach((fieldName, value) -> {
-            ColumnProjection dim = query.getColumnProjection(fieldName);
-            Type<?> fieldType = getType(entityClass, dim);
-            Attribute attribute = projectionToAttribute(dim, fieldType);
+            ColumnProjection columnProjection = query.getColumnProjection(fieldName);
+            Column column = table.getColumn(Column.class, columnProjection.getName());
+
+            Type<?> fieldType = getType(entityClass, columnProjection);
+            Attribute attribute = projectionToAttribute(columnProjection, fieldType);
+
+            ValueType valueType = column.getValueType();
 
             if (entityInstance instanceof ParameterizedModel) {
+
+                //This is an ENUM_TEXT or ENUM_ORDINAL type.
+                if (! fieldType.isEnum()
```

Comment:
separately messaged to clarify this

Reply:
I'm not following.  Do you want me to add more comments on different lines?  What would you like to see?

Reply:
The comment says `This is an ENUM_TEXT or ENUM_ORDINAL type.` but the next line we are doing a negation of `fieldType.isEnum`. That confused me a bit.